src/modules/EventStream.ts

```diff
@@ -0,0 +1,88 @@
+import { FSXARemoteApi, Logger } from './'
+import { NextFunction, Response, Request } from 'express'
+import SSE from 'express-sse-ts'
+import WebSocket from 'ws'
+import { LogLevel } from '..'
+
+const DEFAULT_RECONNECTION_DELAY = 10
+const DEFAULT_MAX_RETRIES = 10
+
+export type EventStreamOptions = {
+  logLevel?: LogLevel
+  reconnectionDelay?: number
+  maxRetries?: number
+}
+
+export class EventStream {
+  api: FSXARemoteApi
+  logger: Logger
+  reconnectionDelay: number
+  maxRetries: number
+  retryCount: number
+  secureToken: string | null = null
+  sse: SSE
+  socket?: WebSocket
+
+  constructor(api: FSXARemoteApi, options?: EventStreamOptions) {
+    this.api = api
+    this.logger = new Logger((options && options.logLevel) || api.logLevel, 'EventStream')
+    this.reconnectionDelay = (options && options.reconnectionDelay) || DEFAULT_RECONNECTION_DELAY
+    this.maxRetries = (options && options.maxRetries) || DEFAULT_MAX_RETRIES
+
+    this.retryCount = 0
+    this.sse = new SSE()
+  }
+
+  async refreshSecureToken() {
+    try {
+      this.secureToken = await this.api.fetchSecureToken()
+    } catch (err) {
+      this.logger.error(err)
+    }
+  }
+
+  async ensureConnection() {
+    if (
+      this.socket?.readyState !== WebSocket.OPEN &&
+      this.socket?.readyState !== WebSocket.CONNECTING
+    ) {
+      if (this.retryCount > this.maxRetries) {
+        this.logger.error(`Can't connect to WebSocket.`)
+        this.sse.send(JSON.stringify({ error: true }))
+      } else {
+        if (!this.secureToken) await this.refreshSecureToken()
```

Comment:
Der Token läuft nach nach ein paar Minuten ab und wird ungültig - muss also in dem Fall erneuert werden.